components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/mediators/RegularExpressionProtector.java

```diff
@@ -83,19 +104,93 @@ public boolean mediate(MessageContext messageContext) {
         if (messageProperty != null) {
             threatType = String.valueOf(messageProperty);
         }
-        checkRequestBody(messageContext);
-        checkRequestHeaders(messageContext);
-        checkRequestPath(messageContext);
+        if (isRequestBodyVulnerable(messageContext) || isRequestHeadersVulnerable(messageContext) ||
+                isRequestPathVulnerable(messageContext)) {
+            return true;
```

Comment:
Is this condition useful since we anyway return true at line 111

Reply:
We need this condition since if a threat has been detected inside the isRequestBodyVulnerable method, there should need to run threat detector sequence. If we removed this it will be removed